File: 141-linked-list-cycle/141-linked-list-cycle.java
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public boolean hasCycle(ListNode head) {
        HashSet<ListNode> trackNodes = new HashSet<ListNode>();
        while(head!=null){
            if(trackNodes.contains(head)){
                return true;
            }
            trackNodes.add(head);
            head=head.next;
        }
        return false;
    }
}

File: 141-linked-list-cycle/141-linked-list-cycle.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def hasCycle(self, head):
        """
        :type head: ListNode
        :rtype: bool
        """
        myDic={}
        
        while head!=None:
            temp=ListNode(head.val, head.next)
            if temp.next in myDic:
                return True
            else:
                myDic[temp.next]=temp.val
            
            head=head.next
                
        return False
            